h/models/group.py

```diff
@@ -41,7 +41,11 @@ class GroupMembership(Base):
     id = sa.Column("id", sa.Integer, autoincrement=True, primary_key=True)
     user_id = sa.Column("user_id", sa.Integer, sa.ForeignKey("user.id"), nullable=False)
     group_id = sa.Column(
-        "group_id", sa.Integer, sa.ForeignKey("group.id"), nullable=False, index=True
+        "group_id",
+        sa.Integer,
+        sa.ForeignKey("group.id", ondelete="cascade"),
```

Comment:
This is about creating a SQL level delete on the FK

"fk__user_group__group_id__group" FOREIGN KEY (group_id) REFERENCES "group"(id)

vs

"fk__user_group__group_id__group" FOREIGN KEY (group_id) REFERENCES "group"(id) ON DELETE CASCADE


You'll need a migration, something like:

```
"""cascade"""
from alembic import op


revision = "f2a76a78f642"
down_revision = "08d3c5a8bd08"


def upgrade():
    op.drop_constraint(
        "fk__user_group__group_id__group", "user_group", type_="foreignkey"
    )
    op.create_foreign_key(
        op.f("fk__user_group__group_id__group"),
        "user_group",
        "group",
        ["group_id"],
        ["id"],
        ondelete="cascade",
    )


def downgrade():
    op.create_foreign_key(
        "fk__user_group__group_id__group", "user_group", "group", ["group_id"], ["id"]
    )
    op.drop_constraint(
        op.f("uq__annotation_metadata__annotation_id"),
        "annotation_metadata",
        type_="unique",
    )

```

Reply:
Added a migration here: https://github.com/hypothesis/h/pull/8713/